web-components/src/components/inputs/new/AmountField/AmountField.TextField.tsx

```diff
@@ -0,0 +1,45 @@
+import { forwardRef } from 'react';
+import { Grid, Theme } from '@mui/material';
+
+import TextField from '../TextField/TextField';
+import { RegenTextFieldProps } from '../TextField/TextField.types';
+
+interface AmountTextFieldProps extends RegenTextFieldProps {
+  availableAmount: number;
+  onMaxClick?: (amount: number) => void;
+}
+
+export const AmountTextField = forwardRef<HTMLDivElement, AmountTextFieldProps>(
+  ({ availableAmount, onMaxClick, ...props }, ref) => {
+    return (
+      <TextField
+        {...props}
+        ref={ref}
+        endAdornment={
+          <Grid
```

Comment:
This would be better indeed, that being said this is outside of the scope of this PR. I'd rather have us fix that in a separate PR to reduce regression risks.